File: Atividade_11/funcionario.py
from abc import ABC, abstractmethod

class Funcionario(ABC):
    @abstractmethod
    def calcular_salario(self):
        pass

class FuncionarioHorista(Funcionario):
    def __init__(self, horas_trabalhadas, valor_hora):
        self.horas_trabalhadas = horas_trabalhadas
        self.valor_hora = valor_hora

    def calcular_salario(self):
        return self.horas_trabalhadas * self.valor_hora

class FuncionarioAssalariado(Funcionario):
    def __init__(self, salario_fixo):
        self.salario_fixo = salario_fixo

    def calcular_salario(self):
        return self.salario_fixo

# Testando
horista = FuncionarioHorista(160, 25)
assalariado = FuncionarioAssalariado(3000)

print(f"Salário do horista: R$ {horista.calcular_salario()}")
print(f"Salário do assalariado: R$ {assalariado.calcular_salario()}")

File: Atividade_11/funcionario.go
package main

import "fmt"


type Funcionario interface {
	CalcularSalario() float64
}


type FuncionarioHorista struct {
	horasTrabalhadas int
	valorHora        float64
}

func (f FuncionarioHorista) CalcularSalario() float64 {
	return float64(f.horasTrabalhadas) * f.valorHora
}


type FuncionarioAssalariado struct {
	salarioFixo float64
}

func (f FuncionarioAssalariado) CalcularSalario() float64 {
	return f.salarioFixo
}

func main() {
	horista := FuncionarioHorista{160, 25}
	assalariado := FuncionarioAssalariado{3000}

	fmt.Printf("Salário do horista: R$ %.2f\n", horista.CalcularSalario())
	fmt.Printf("Salário do assalariado: R$ %.2f\n", assalariado.CalcularSalario())
}
